Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -157,10 +157,6 @@ WarpXParticleContainer::AddNParticles (int /*lev*/,
         p.pos(1) = z[i];
 #endif
 
-        if ( (NumRuntimeRealComps()>0) || (NumRuntimeIntComps()>0) ){
-            auto& ptile = DefineAndReturnParticleTile(0, 0, 0);
-        }
-
```

Comment:
Fixes the warning:
`./Source/Particles/WarpXParticleContainer.cpp:161:19`: warning: unused variable ‘ptile’ [-Wunused-variable]
             auto& ptile = DefineAndReturnParticleTile(0, 0, 0);

Reply:
I am not sure we can do that, the output variable is not used but we may need to call `DefineAndReturnParticleTile`. I will confirm with @atmyers. Same thing below.

Reply:
we do need to call this - one thing that might work to silence the warning is to wrap the call in `amrex::ignore_unused`, or possibly just not just capture the return value at all.

Reply:
Just to clarify, @atmyers do you mean something like
```c++
(void) DefineAndReturnParticleTile(0, 0, 0);
```
or more modern
```c++
static_cast<void> DefineAndReturnParticleTile(0, 0, 0);
```
?

Reply:
I mean either:

`amrex::ignore_unused(DefineAndReturnParticleTile(0, 0, 0));`

or maybe just

`DefineAndReturnParticleTile(0, 0, 0);`

not sure if the second one will avoid the warning or not.

Reply:
Actually the latter should do, just discard the return value.
As long as special properties are not set on that function, this will not throw a warning.

Reply:
Thanks @MaxThevenet and @atmyers for pointing this out and thanks @ax3l for suggesting a solution that does not trigger a warning. I have one small question left: if we do need to call this function, why did all the tests pass when I removed the function call?

Reply:
I am not sure exactly why here, but in general I think it is because we don't have enough tests, so either we didn't hit exactly this piece of code or maybe that would generate a random error and it randomly passed.